packages/fe-fpm-writer/test/unit/filter.test.ts

```diff
@@ -0,0 +1,304 @@
+import { Placement } from '../../src/common/types';
+import os from 'os';
+
+describe('CustomFilter', () => {
+    describe('generateCustomFilter', () => {
+        const testDir = '' + Date.now();
+        let fs: Editor;
+        const filter: CustomFilter = {
+            name: 'NewCustomFilter',
+            label: 'Test Custom Filter',
+            controlID: 'testID',
+            property: 'Testing'
+        };
+        const getControllerPath = (controller: CustomFilter, isTs = false): string => {
+            return join(testDir, `webapp/${controller.folder}/${controller.name}.${isTs ? 'ts' : 'js'}`);
+        };
+        const getExpectedFragmentPath = (newFilter: CustomFilter): string =>
+            join(testDir, `webapp/${newFilter.folder}/${newFilter.fragmentFile}.fragment.xml`);
+
+        const getManifest = (extensions: unknown = undefined): string => {
+            const manifest = {
+                'sap.app': {
+                    id: 'myTestApp'
+                },
+                'sap.ui5': {
+                    dependencies: {
+                        libs: {
+                            'sap.fe.templates': {}
+                        }
+                    },
+                    routing: {
+                        targets: {
+                            TestListReport: { name: 'sap.fe.templates.ListReport' },
+                            TestObjectPage: { name: 'sap.fe.templates.ObjectPage' }
+                        }
+                    }
+                }
+            };
+            return JSON.stringify(manifest, null, 2);
+        };
+        const testAppManifest = getManifest();
+
+        beforeEach(() => {
+            fs = create(createStorage());
+            fs.delete(testDir);
+            fs.write(join(testDir, 'webapp/manifest.json'), testAppManifest);
+        });
+        test('New custom filter (no eventhandler)', () => {
+            generateCustomFilter(testDir, filter, fs);
+            expect(fs.readJSON(join(testDir, 'webapp/manifest.json'))).toMatchSnapshot();
+            expect(fs.exists(getControllerPath(filter))).toBeTruthy();
+            expect(fs.read(getControllerPath(filter))).toMatchSnapshot();
+            expect(fs.exists(getExpectedFragmentPath(filter))).toBeTruthy();
+            expect(fs.read(getExpectedFragmentPath(filter))).toMatchSnapshot();
+        });
+
+        test('Create several new custom filters', () => {
+            const secondFilter = {
+                name: 'NewCustomFilter2',
+                label: 'Test Custom Filter 2',
+                controlID: 'testID2',
+                property: 'Testing'
+            };
+            generateCustomFilter(testDir, filter, fs);
+
+            generateCustomFilter(testDir, secondFilter, fs);
+            expect(fs.readJSON(join(testDir, 'webapp/manifest.json'))).toMatchSnapshot();
+            expect(fs.exists(getControllerPath(secondFilter))).toBeTruthy();
```

Comment:
This test is incorrect since it expects a new controller despite the input being `undefined`

Reply:
Fixed with #1027